src/main/java/com/gooddata/warehouse/WarehouseS3Credentials.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2007-2017, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.warehouse;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.gooddata.util.ISODateTimeDeserializer;
+import com.gooddata.util.ISODateTimeSerializer;
+import org.joda.time.DateTime;
+import org.springframework.web.util.UriTemplate;
+
+import java.util.Map;
+
+import static com.gooddata.util.Validate.notNullState;
+
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonTypeName("s3Credentials")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class WarehouseS3Credentials {
+
+    public static final String URI = WarehouseS3CredentialsList.URI + "/{region}/{accessKey}";
+    public static final UriTemplate TEMPLATE = new UriTemplate(URI);
+    private static final String SELF_LINK = "self";
+
+    private final DateTime createdAt;
+
+    private final String userLogin;
+
+    private final String secretKey;
+
+    private final String accessKey;
+
+    private final String region;
+
+    private Map<String, String> links;
+
+    /**
+     * Used to create/update S3 credentials via POST/PUT
```

Comment:
POST/PUT wording is confusing, remove.
create/update wording is confusing as well, there is only add method